controls/V-81019.rb

```diff
@@ -0,0 +1,51 @@
+# encoding: utf-8
+#
+control "V-81019" do
+  title "The Red Hat Enterprise Linux operating system must take appropriate action when the audisp-remote buffer is full."
+  desc  "
+   Information stored in one location is vulnerable to accidental or incidental deletion or alteration. Off-loading is a 
+   common process in information systems with limited audit storage capacity. When the remote buffer is full, audit 
+   logs will not be collected and sent to the central log server.
+  "
+  impact 0.5
+  tag "gtitle": "SRG-OS-000342-GPOS-00133"
+  tag "satisfies": ["SRG-OS-000342-GPOS-00133", "SRG-OS-000479-GPOS-00224"]
+  tag "gid": "V-81019"
+  tag "rid": "SV-95731r1_rule"
+  tag "stig_id": "RHEL-07-030210"
+  tag "cci": ["CCI-001851"]
+  tag "documentable": false
+  tag "nist": ["AU-12 c", "Rev_4"]
+  tag "subsystems": ["audit"]
+  tag "check_id": "C-80735r1_chk"
+  tag "fix_id": "F-87853r3_fix"
+  desc "check", "
+  Verify the audisp daemon is configured to take an appropriate action when the internal queue is full:
+
+  # grep \"overflow_action\" /etc/audisp/audispd.conf
+
+  overflow_action = syslog
+
+  If the \"overflow_action\" option is not \"syslog\", \"single\", or \"halt\", or the line is commented out, this is a finding.
+  "
+  desc "fix", "
+  Edit the /etc/audisp/audispd.conf file and add or update the \"overflow_action\" option:
+
+  overflow_action = syslog
+
+  The audit daemon must be restarted for changes to take effect:
+
+  # service auditd restart
+  "
+
+  if file('/etc/audisp/audispd.conf').exist?
+    describe parse_config_file('/etc/audisp/audispd.conf') do
+      its('overflow_action') { should match %r{syslog$|single$|halt$} }
```

Comment:
It would be more clear to use arrays here vs regex.

```ruby
its('overflow_action') { should be_in %w(syslog single halt) }
```

Reply:
I think this control also needs to check if the line is commented out.

Reply:
How does the `parse_config_file` resource act when the line is commented out and you attempt to match a value? Does it just return null? What the STIG does not say explicitly is that if it is not present at all this is a finding but I'd say we can infer that from the spirit of this control. Therefore, if checking for a value `its('param')` returns null we can assume that it is either some other value, it's commented out or simply not present. In any case the output will likewise indicate its actual value or the fact that it is null. All of which are findings. I don't see a reason to check for specific cases when we've already effectively accounted for them.